### PR TITLE
[plugin.video.filmarkivet] 3.0.0

### PR DESCRIPTION
**Pull request by pugo**
### Description
- Complete rewrite of the plugin to match the new updated source site (filmarkivet.se).
- Removed several dependencies and new code is much cleaner and smaller.
- Added caching for speed-ups.
### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a sinlge commit with using the following style: [plugin.video.foo] v1.0.0

Addtional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuesly monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.filmarkivet/addon.xml [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+    <import addon="xbmc.python" version="2.14.0"/>
+    <import addon="script.module.beautifulsoup4" version="4.3.2"/>
   </requires>
   <extension point="xbmc.python.pluginsource" library="default.py">
     <provides>video</provides>
   </extension>
   <extension point="xbmc.addon.metadata">
     <platform>all</platform>
-    <language>se</language>
+    <language>sv en de</language>
```
Comment (phil65) [inline plugin.video.filmarkivet/addon.xml]: the site provides swedish, english and german movies?

File: plugin.video.filmarkivet/addon.xml [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
+    <email>pugo@pugo.org</email>
+    <license>GNU General Public License, v3</license>
+    <source>https://github.com/search?q=filmarkivet</source>
```
Comment (phil65) [inline plugin.video.filmarkivet/addon.xml]: a github search as source? :)

File: plugin.video.filmarkivet/lib/filmarkivet.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+		except:
+			return None, None
+
+	def get_url_movies(self, url, mode, page=1, limit=False):
+		get_url = url
+		if limit:
+			get_url += '{}limit={}&pg={}'.format('?' if url.rfind('?') < 0 else '&', self.MOVIES_PER_PAGE, page)
+		html = self.webget.getURL(get_url)
+		soup = BeautifulSoup(html, 'html.parser')
+		range, range_max = self.__get_range(soup)
```
Comment (phil65) [inline plugin.video.filmarkivet/lib/filmarkivet.py]: better do not use names of builtin functions as var names (range..)

File: plugin.video.filmarkivet/resources/language/English/strings.xml [hunk 4]
```diff
@@ -2,16 +2,13 @@
 <strings>
```
Comment (phil65) [inline plugin.video.filmarkivet/resources/language/English/strings.xml]: Would be cool if you could convert the language xmls to po files at some point.
(no need to it with this PR)

Comment (pugo) [inline plugin.video.filmarkivet/resources/language/English/strings.xml]: Ok. Thanks for input! :)

File: plugin.video.filmarkivet/lib/filmarkivet.py (hunk 3 above)
Comment (pugo) [inline plugin.video.filmarkivet/lib/filmarkivet.py]: Very true. Thanks. Will update the PR tomorrow.

**Comment by pugo** [comment]
Pull request updated. Do I need to do anything more than just update?

**Comment by phil65** [comment]
nope, all fine.  Thx!

**Pull request merged**